BLL/Services/Implementations/CategoryService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using BLL.DTO.Request.Category;
+using BLL.DTO.Response;
+using BLL.Service.Interface;
+using DAL.Model;
+using DAL.Repositories.Interfaces;
+using BLL.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AutoMapper;
+
+namespace BLL.Services.Implementations
+{
+    public class CategoryService : ICategoryService
+    {
+        private readonly ICategoryRepository _categoryRepository;
+        private readonly IMapper _mapper;
+
+        public CategoryService(ICategoryRepository categoryRepository, IMapper mapper)
+        {
+            _categoryRepository = categoryRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<CategoryDto>> GetAllCategoriesAsync(CancellationToken cancellationToken = default = default)
+        {
+            var categories = await _categoryRepository.GetAllAsync(cancellationToken);
+            return _mapper.Map<IEnumerable<CategoryDto>>(categories);
+        }
+
+        public async Task<CategoryDto> GetCategoryByIdAsync(Guid categoryId, CancellationToken cancellationToken = default = default)
+        {
+            var category = await _categoryRepository.GetByIdAsync(categoryId, cancellationToken);
+            if (category is null)
+                throw new EntityNotFoundException($"Category with Id {categoryId} not found.");
+
+            return _mapper.Map<CategoryDto>(category);
+        }
+
+        public async Task AddCategoryAsync(CreateCategoryDto categoryDto, CancellationToken cancellationToken = default = default)
+        {
+            var category = _mapper.Map<Category>(categoryDto);
+            category.Id = Guid.NewGuid();
+
```

Comment:
it is better to add a check before creation, since a category with that name may already exist